### PR TITLE
feat(storagetransfer): add quickstart for storage transfer service

### PR DESCRIPTION
**Pull request by JesseLovelace**
Adds a quickstart for storage transfer service, along with a test

I have never written Go code before (ever), so apologies for any mistakes!

### CODE REVIEW

File: storagetransfer/quickstart_test.go
Comment (tritone) [inline storagetransfer/quickstart_test.go:66]: I don't love doing this with a regex from output-- could you have the quickstart function return the job name (or something containing it) instead?

File: storagetransfer/quickstart_test.go
Comment (tritone) [inline storagetransfer/quickstart_test.go:50]: If I understand correctly, both of these buckets are empty. Does that pose any problems for testing the transfer job, or does it not matter?

File: storagetransfer/quickstart_test.go
Comment (tritone) [inline storagetransfer/quickstart_test.go:86]: Possibly totally off-base but-- should this code really be in a sample as well, rather than just in the test? Do the docs make it clear how to give these permissions to the STS service account?

**Review by tritone — COMMENTED** [review]
Welcome to Go! Few comments, some of which are because I don't know anything about STS.

Also, to fix the build, you first need to run `go get cloud.google.com/go/storagetransfer` and probably `go mod tidy` as well. Then we can see if there are other things to fix.

File: storagetransfer/quickstart.go
Comment (JesseLovelace) [reply storagetransfer/quickstart.go:14]: Probably not super typical, but a valid use case and the simplest thing we could think of for a get-off-the-ground quickstart

File: storagetransfer/quickstart.go
Comment (JesseLovelace) [reply storagetransfer/quickstart.go:47]: The customer would only need JobName to monitor the job, which they already have

File: storagetransfer/quickstart_test.go
Comment (JesseLovelace) [reply storagetransfer/quickstart_test.go:50]: It doesn't matter, the job creates and runs successfully anyway, doing nothing

File: storagetransfer/quickstart_test.go
Comment (JesseLovelace) [reply storagetransfer/quickstart_test.go:86]: The docs explain how to do it in the console, but not with code. I agree that it should be a sample and it's something I want to add and will propose at some point, just not sure how to go about it yet since it's done entirely with the GCS client and not with the STS client, so there has to be some explainer on how to depend on the GCS client as well, I'm not sure if there's other examples of docs in one client having samples for a different client. Either way, it's not going to be part of the this stage

**Comment by JesseLovelace** [comment]
Addressed your comments, however running `go get cloud.google.com/go/storagetransfer` didn't seem to do anything

File: storagetransfer/quickstart.go
Comment (tritone) [inline storagetransfer/quickstart.go:9]: Core language dependencies should come first; run goimports to fix (intellij can do this for you).

File: storagetransfer/quickstart_test.go
Comment (tritone) [inline storagetransfer/quickstart_test.go:25]: This and L31 should be `t.Fatalf` instead of log.

**Review by tritone — COMMENTED** [review]
Two more little things, otherwise LGTM once we fix the build.

File: storagetransfer/quickstart_test.go
Comment (tritone) [inline storagetransfer/quickstart_test.go:82]: This also needs to cause an error rather than proceeding. You should either make the function return an error to the caller, or pass in the `testing.T` as an arg so that you can cause the test failure here.

**Review by tritone — APPROVED** [review]
LGTM pending presubmits passing. @codyoss we need a review from you as well because we don't have ownership over this subdir.

**Review by codyoss — APPROVED** [review]

**Comment by JesseLovelace** [comment]
Seems like this is just pending https://github.com/GoogleCloudPlatform/golang-samples/issues/2300

**Comment by tritone** [comment]
Yeah, seems to be consistently failing. I made #2308 to fix.

**Pull request merged**
